src/plugins/data_source/config.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema, TypeOf } from '@osd/config-schema';
+import { fileAppenderSchema } from './audit_config';
+
+const KEY_NAME_MIN_LENGTH: number = 1;
+const KEY_NAME_MAX_LENGTH: number = 100;
+// Wrapping key size shoule be 32 bytes, as used in envelope encryption algorithms.
+const WRAPPING_KEY_SIZE: number = 32;
+
+export const configSchema = schema.object({
+  enabled: schema.boolean({ defaultValue: false }),
+  encryption: schema.object({
+    wrappingKeyName: schema.string({
+      minLength: KEY_NAME_MIN_LENGTH,
+      maxLength: KEY_NAME_MAX_LENGTH,
+      defaultValue: 'changeme',
+    }),
+    wrappingKeyNamespace: schema.string({
+      minLength: KEY_NAME_MIN_LENGTH,
+      maxLength: KEY_NAME_MAX_LENGTH,
+      defaultValue: 'changeme',
+    }),
+    wrappingKey: schema.arrayOf(schema.number(), {
+      minSize: WRAPPING_KEY_SIZE,
+      maxSize: WRAPPING_KEY_SIZE,
+      defaultValue: new Array(32).fill(0),
+    }),
+  }),
+  clientPool: schema.object({
+    size: schema.number({ defaultValue: 5 }),
+  }),
+  audit: schema.object({
```

Comment:
nit: how come we want enable multiple log files? would there be a reason to just utilize existing auditing functionality?

Reply:
audit log is different with service log, for data source we'll have both. You can check #1986  for more discussion. cc @kristenTian